### PR TITLE
MGMT-2611 - Event on failing validation specify the validation

### PR DESCRIPTION
**Pull request by oshercc**

### CODE REVIEW

File: internal/host/common.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ const (
 	statusInfoDiscovering                    = "Waiting for host to send hardware details"
 	statusInfoInsufficientHardware           = "Host does not meet the minimum hardware requirements"
 	statusInfoPendingForInput                = "Waiting for user input"
-	statusInfoNotReadyForInstall             = "Host cannot be installed due to failing validation(s)"
+	statusInfoNotReadyForInstall             = "Host cannot be installed due to fallowing failing validation(s): $VALIDATIONS"
```
Comment (ori-amizur) [inline internal/host/common.go:33]: Typo fallowing - following

File: internal/host/transition.go [hunk 2]
```diff
@@ -501,3 +506,20 @@ func (th *transitionHandler) IsAddToExistingClusterHost(sw stateswitch.StateSwit
 	}
 	return swag.StringValue(sHost.host.Kind) == models.HostKindAddToExistingClusterHost, nil
 }
+
+func getFailedValidations(b []byte, params *TransitionArgsRefreshHost) []string {
+	var dat map[string][]validationResult
+
+	if err := json.Unmarshal(b, &dat); err != nil {
+		panic(err)
```
Comment (ori-amizur) [inline internal/host/transition.go:514]: Why panic?

File: internal/host/common.go (hunk 1 above)
Comment (ori-amizur) [inline internal/host/common.go:33]: Suggest to change to $FAILING_VALIDATIONS

File: internal/host/common.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ const (
 	statusInfoDiscovering                    = "Waiting for host to send hardware details"
 	statusInfoInsufficientHardware           = "Host does not meet the minimum hardware requirements"
```
Comment (ori-amizur) [inline internal/host/common.go:31]: You might want to do the same for statusInfoInsufficientHardware and statusInfoPendingForInput

File: internal/host/transition.go [hunk 4]
```diff
@@ -501,3 +506,20 @@ func (th *transitionHandler) IsAddToExistingClusterHost(sw stateswitch.StateSwit
 	}
 	return swag.StringValue(sHost.host.Kind) == models.HostKindAddToExistingClusterHost, nil
 }
+
+func getFailedValidations(b []byte, params *TransitionArgsRefreshHost) []string {
+	var dat map[string][]validationResult
+
+	if err := json.Unmarshal(b, &dat); err != nil {
```
Comment (ori-amizur) [inline internal/host/transition.go:513]: Why is the dat needed? It is not used

File: internal/host/common.go (hunk 3 above)
Comment (oshercc) [reply internal/host/common.go:31]: added separate ticket: https://issues.redhat.com/browse/MGMT-2831

File: internal/host/transition_test.go [hunk 5]
```diff
@@ -1282,6 +1282,7 @@ var _ = Describe("Refresh Host", func() {
 				srcState:          models.HostStatusDiscovering,
 				dstState:          models.HostStatusDisconnected,
 				statusInfoChecker: makeValueChecker(statusInfoDisconnected),
+				//statusInfoChecker: makeJsonChecker(statusInfoDisconnected),
```
Comment (ori-amizur) [inline internal/host/transition_test.go:1285]: Why the comment?

File: internal/host/transition_test.go [hunk 6]
```diff
@@ -1631,7 +1635,7 @@ var _ = Describe("Refresh Host", func() {
 				dstState:           models.HostStatusInsufficient,
 				machineNetworkCidr: "1.2.3.0/24",
 				role:               models.HostRoleMaster,
-				statusInfoChecker:  makeValueChecker(statusInfoNotReadyForInstall),
+				statusInfoChecker:  makeValueChecker(formatStatusInfoNotReadyForInstall("Require at least 4 CPU cores for master role, found only 2, Require at least 16 GiB RAM role master, found only 8")),
```
Comment (ori-amizur) [inline internal/host/transition_test.go:1638]: Seems that when there are multiple messages each containing a comma "," then it is hard to distinguish which is the message

Comment (avishayt) [reply internal/host/transition_test.go:1638]: Agree.  Maybe join with semi-colon?

**Review by avishayt — APPROVED** [review]

**Pull request merged**